src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+
+namespace SFA.DAS.AANHub.Application.CalendarEvents.Queries;
+
+public class GetCalendarEventsQueryHandler : IRequestHandler<GetCalendarEventsQuery, ValidatedResponse<GetCalendarEventsQueryResult>>
+{
+    private readonly ICalendarEventsReadRepository _calendarEventsReadRepository;
+
+    public GetCalendarEventsQueryHandler(ICalendarEventsReadRepository calendarEventsReadRepository)
+    {
+        _calendarEventsReadRepository = calendarEventsReadRepository;
+    }
+
+    public async Task<ValidatedResponse<GetCalendarEventsQueryResult>> Handle(GetCalendarEventsQuery request, CancellationToken cancellationToken)
+    {
+        var pageSize = Domain.Common.Constants.CalendarEvents.PageSize;
+        var page = 1;
+        var response =
+            await _calendarEventsReadRepository.GetCalendarEvents(request.RequestedByMemberId, cancellationToken);
+
+        var result = new GetCalendarEventsQueryResult
+        {
+            Page = page,
+            PageSize = pageSize,
+            TotalCount = response.Count,
+            TotalPages = (response.Count + pageSize - 1) / pageSize,
```

Comment:
I am getting confused due to half baked paging functionality. Just leave total pages to zero as you are returning all the pages. Also do not return page size in the response. 
On top the correct way of calculating TotalPages would be to get a full count first then use `(decimal)Math.Ceiling(response.First().TotalCount / (decimal)pageSize`

Reply:
ok set totalpages and pagesize to 0 until we put pagination in